cmd/config_test.go

```diff
@@ -29,7 +33,7 @@ func TestParseEnvURLStr(t *testing.T) {
 		port         string
 	}{
 		{
-			hostURL:   "https://minio:minio1#23@localhost:9000",
+			hostURL:   "https://minio:minio1%2323@localhost:9000",
```

Comment:
This is not correct users won't pass URL encoded passwords in this manner @vadmeste and they are not expected to.

Reply:
> This is not correct users won't pass URL encoded passwords in this manner @vadmeste and they are not expected to.

Basically, this change is needed if the access or secret keys contains '@' or ':'

I think MinIO or AWS has # or @, this should not be a problem. But one user seems to have picked a key with '@' in it and that made an issue with MC_HOST_ parsing in mc code.

Reply:
The current fix is wrong because you are expecting users to provide an URL encoded form, they will never be able to provide that as input. 

I think you can simply ignore that user they avoid creating passwords in that manner - I don't think this PR is needed.